producers/mysql-producer/src/main/java/com/flipkart/aesop/runtime/producer/eventprocessor/impl/InsertEventV2Processor.java

```diff
@@ -33,10 +38,18 @@ public void process(BinlogEventV4 event, OpenReplicationListener listener) throw
 		}
 		LOGGER.debug("Insert Event Received : " + event);
 		WriteRowsEventV2 wre = (WriteRowsEventV2) event;
-		listener.getMysqlTransactionManager().performChanges(wre.getTableId(), wre.getHeader(), wre.getRows(),
-		        DbusOpcode.UPSERT);
+		List<Row> rowList = wre.getRows();
+		List<Pair<Row>> listOfPairs = new ArrayList<Pair<Row>>(rowList.size());
```

Comment:
If possible, refactor the diff calculation for all event processors (except Update).